cookbooks/bcpc-hadoop/recipes/os_basic_queries.rb

```diff
@@ -0,0 +1,75 @@
+# Get a list of cluster nodes to monitor
+monitored_nodes_objs = get_all_nodes.map { |n| n if not n.hostname =~ /bootstrap/ }.compact
+
+# Graphite queries which specify property to query and alarming trigger, severity(maps to zabbix's api -> trigger -> priority)
+# and owner who the trigger is routed to for resolution
+# Queries are structured as they appear in graphite dashboard
+# Refer Zabbix API(https://www.zabbix.com/documentation/2.2/manual/api/reference/) for the various attributes used
+# The following list is supposed to be a skeleton which should be extended/edited according to the deployment scenario
+# By default items, triggers actions are disabled and won't be monitored. To enable them, set "enable=true" in the json structure
+monitored_nodes_objs.each do |node_obj|
+  host = node_obj.hostname
+  node.default["bcpc"]["hadoop"]["graphite"]["queries"]["#{host}"] = { 
+      "memory_active_#{host}" => {
+        'type' => "servers",
+        'query' => "memory.Active",
+        'trigger_val' => "max(61,0)",
+        'trigger_cond' => "=0",
+        'trigger_name' => "#{host}_NodeAvailability",
+        'enable' => true,
+        'trigger_desc' => "Node seems to be down",
+        'severity' => 5,
+        'route_to' => "admin"
+      },
+      "chef_client_success_#{host}" => {
+        'type' => "chef",
+        'query' => "success",
+        'trigger_val' => "max(#{((node['chef_client']['interval'].to_i + node['chef_client']['splay'].to_i) / 30).ceil}m, 0)",
```

Comment:
The arithmetic here is confusing to me; what are you doing?
` 'trigger_val' => "max(#{((node['chef_client']['interval'].to_i + node['chef_client']['splay'].to_i) / 30).ceil}m, 0)",`